app/src/main/java/com/kidssaveocean/fatechanger/firebase/FirebaseService.kt

```diff
@@ -0,0 +1,80 @@
+package com.kidssaveocean.fatechanger.firebase
+
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+import java.util.*
+
+class FirebaseService : Observable ()  {
```

Comment:
If this is being treated as a singleton, you can optionally make this class into an object and remove the companion object from inside it

Reply:
Yeah, actually, it was a singelton, but I couldn't keep private constructor with Observable() and I just removed it and forgot about that.

Reply:
I didn't understand what you meant by
> make this class into an object
So, I haven't change anything here for now. I'll change it in the future when I know more about architectural pattern in Android for this case and what is the most appropriate pattern here.